app/Services/GitHub/GitHubClientFactory.php

```
@@ -10,23 +11,84 @@
 
 class GitHubClientFactory implements GitHubClientFactoryContract
 {
-    public function __construct(private GitHubFactory $factory) {}
+    public function __construct(
+        private GitHubFactory $factory,
+        private GitHubConnectionManager $connections,
+    ) {}
 
     public function make(?string $connection = null, bool $withCache = true): Client
```

PHPMD (2.15.0)

The method make has a boolean flag argument $withCache, which is a certain sign of a Single Responsibility Principle violation. (undefined)

(BooleanArgumentFlag)